### PR TITLE
rewrite to dynamically generate SVG components (for Deku & React)

### LINKED ISSUE #24: should be a repository of svgs with a single js entrypoint
@anthonyshort should be able to export .svgs into here without having to touch css or js

probably requires a pre-compile step for finding all svgs and converting them to jsx things

not a huge priority, but doing this would prevent me from ever having to commit here again

### PR DESCRIPTION
**Pull request by stephenmathieson**
This is a breaking change. The exported API is completely different. Rather than exporting a higher-order `Icon` component, we just export the individual components. This enables us to keep our bundle size down (as long as we're tree-shaking) by only importing the things we need. The new API looks something like:

```js
import React from 'react'
import { Alert } from '@segment/icons/react'

export default () => (
  <div className='whatever'>
    <Alert color='red' />
  </div>
)
```

or

```js
import 'element' from 'virtual-element'
import { Alert } from '@segment/icons/deku'

export default { render }

function render () {
  <div class='whatever'>
    <Alert color='red' />
  </div>
}
```

NOTE: I've dropped support for the `size=` attribute, as this should be the responsibility of the parent component (not the icon itself). This could be added back with little-to-no effort, but it'll take some convincing :P

NOTEx2: you're now able to use `__COLOR__` for color dynamic support in your SVGs.

NOTEx3: I moved the codebase to `standard` rather than the custom eslint rules no one cares for.

---

The `build.js` script will now take any `.svg` files in the `./svgs` directory and output both React and Deku components for each. This should make design -> eng handoffs much easier, as engineers (and/or designers) are now able to export SVGs from Sketch into this repo and run `make` (closes #24). 🎉 

This diff is huge, so I've separated the work into multiple (self documenting) commits for easier review.

This is not "complete", as I'd like to do a few more things (which I'll likely get to in the morning, since I'm blocked on my normal Sprint work):

- [x] ~~include sourcemaps in the "built" components~~ decided against this; not worth the work
- [x] further optimize the SVGs at build-time (include more SVGO plugins)
- [x] export PascalCase components rather than camelCase
- [x] include `displayName` property in React components
- [x] add back `hoverColor` support
- [x] add basic "it can render" tests for both runtimes

Please review @segmentio/site @anthonyshort

### CODE REVIEW

**Comment by benjaminhoffman** [comment]
👏 👏 👏 

question:
1) i dont see any corresponding changes in app or site-\* repos.... how do we reconcile this breaking change?

2) you mention size is now responsibility of parent component.  doesn't it make more sense for the size to be passed down as a prop to the icon component?  and, can you give an example of the new way... you mean something like this:

``` js
import { alert } from '@segment/icons/deku'
...
export default () => (
  <div className='whatever' size={'12px'}>
    <alert color='red' />
  </div>
)
```

**Comment by stephenmathieson** [comment]
> i dont see any corresponding changes in app or site-\* repos.... how do we reconcile this breaking change?

We'll bump major here and the other repos will pull this change in when they're ready. Hooray semver!

> you mention size is now responsibility of parent component. doesn't it make more sense for the size to be passed down as a prop to the icon component? and, can you give an example of the new way...

We'll just use CSS rather than inline styles:

**MyComponent.js**

``` js
import { alert } from '...'
import styles from './MyComponent.css'

export default () => (
  <div className={styles.whatever}>
    <alert color='red' />
  </div>
)
```

**MyComponent.css**

``` css
.whatever {
  width: 12px;
  height: 12px;
}
```

Could also use inline styles, but generally speaking, CSS is better.

File: templates/react.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  };
+
+  constructor () {
+    super()
+    this.displayName = '__NAME__'
+    this.state = { isHovering: false }
+  }
+
+  render () {
+    const color = this.state.isHovering ? this.props.hoverColor : this.props.color
```
Comment (Rowno) [inline templates/react.js]: If all the icons were set to `currentColor` you would easily be able to change their color from the parent component using the CSS `color` property. It also has the benefit that the icons will automatically inherit the surrounding text color (including it's hover/focus/active color). 😉

Comment (Rowno) [inline templates/react.js:17]: If all the icons were set to `currentColor` you would easily be able to change their color from the parent component using the CSS `color` property. It also has the benefit that the icons will automatically inherit the surrounding text color (including it's hover/focus/active color). 😉

**Review by Rowno — DISMISSED** [review]

File: templates/react.js (hunk 1 above)
Comment (stephenmathieson) [reply templates/react.js:17]: Interesting. I'm on my way to the train now. Let's chat when I get in.

File: package.json [hunk 2]
```diff
@@ -2,29 +2,41 @@
...
+    "mz": "^2.5.0",
+    "pascal-case": "^2.0.0",
+    "react-dom": "^15.3.2",
+    "standard": "^8.5.0",
+    "svgo": "^0.7.1"
   },
   "dependencies": {
-    "deku": "0.5.6",
-    "magic-virtual-element": "1.0.6",
+    "react": "^15.3.2",
```
Comment (timhudson) [inline package.json:31]: This should probably be a `devDependency` and `peerDependency`.

File: svgs/warning-status.svg [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<svg width="14px" height="14px" viewBox="0 0 14 14">
```
Comment (timhudson) [inline svgs/warning-status.svg]: Any reason this isn't using `width="100%" height="100%"` like the others?

**Review by timhudson — DISMISSED** [review]

File: package.json (hunk 2 above)
Comment (stephenmathieson) [reply package.json:31]: Yep, you're right. Will update.

File: svgs/warning-status.svg (hunk 3 above)
Comment (stephenmathieson) [reply svgs/warning-status.svg]: Not sure, all I did here was move it from `./src/` to `./svgs`.

Can update!

**Comment by stevenmiller888** [comment]
@stephenmathieson Can we get this in?

File: templates/react.js (hunk 1 above)
Comment (stephenmathieson) [reply templates/react.js:17]: Let's mark this as a TODO, as we need this to land before next sprint

**Comment by stephenmathieson** [comment]
YOLO merging

@timhudson @Rowno LMK if anything needs to change

**Pull request merged**